SafeExamBrowser.UserInterface.Desktop/Controls/Taskbar/NetworkControl.xaml.cs

```diff
@@ -45,22 +45,53 @@ private void InitializeWirelessNetworkControl()
 			var originalBrush = Button.Background;
 
 			adapter.Changed += () => Dispatcher.InvokeAsync(Update);
+			var lastOpenedBySpacePress = false;
+			Button.PreviewKeyDown += (o, args) =>
+			{
+				if (args.Key == System.Windows.Input.Key.Space)					// for some reason, the popup immediately closes again if opened by a Space Bar key event - as a mitigation, we record the space bar event and leave the popup open for at least 3 seconds
+				{
+					lastOpenedBySpacePress = true;
+				}
+			};
 			Button.Click += (o, args) => Popup.IsOpen = !Popup.IsOpen;
-			Button.MouseLeave += (o, args) => Task.Delay(250).ContinueWith(_ => Dispatcher.Invoke(() => Popup.IsOpen = Popup.IsMouseOver));
+			Button.MouseLeave += (o, args) => Task.Delay(250).ContinueWith(_ => Dispatcher.Invoke(() =>
+			{
+				if (Popup.IsOpen && lastOpenedBySpacePress)
+				{
+					return;
+				}
+				Popup.IsOpen = Popup.IsMouseOver;
+			}));
 			Popup.CustomPopupPlacementCallback = new CustomPopupPlacementCallback(Popup_PlacementCallback);
-			Popup.MouseLeave += (o, args) => Task.Delay(250).ContinueWith(_ => Dispatcher.Invoke(() => Popup.IsOpen = IsMouseOver));
+			Popup.MouseLeave += (o, args) => Task.Delay(250).ContinueWith(_ => Dispatcher.Invoke(() =>
+			{
+				if (Popup.IsOpen && lastOpenedBySpacePress)
+				{
+					return;
+				}
+				Popup.IsOpen = IsMouseOver;
+			}));
 			WirelessIcon.Child = GetWirelessIcon(0);
 
 			Popup.Opened += (o, args) =>
 			{
 				Background = Brushes.LightGray;
 				Button.Background = Brushes.LightGray;
+				Task.Delay(100).ContinueWith((task) => Dispatcher.Invoke(() =>
+				{
+					var btn = WirelessNetworksStackPanel.Children[0] as NetworkButton;
```

Comment:
Same here regarding `IndexOutOfRangeException`, see comment above.

Reply:
Thanks for the hint, this is now fixed.